### PR TITLE
(#26) Single search result redirects to definition

### LINKED ISSUE #26: Story: Single Result Returned from Search
## As a user, if my search term only returns one result, I would like to be redirected to the term’s page instead of the results page so that I may view all relevant details for the term matched without needing to click the results in the list to get to the term page.

> **ESTIMATE** 3

### Acceptance Criteria

```gherkin
Scenario: As a user, if my search term only returns one result, I would like to be redirected to the term’s page instead of the results page. 
  Given user is on landing Cancer Terms page 
  When user starts typing "lung cancer" in the search bar 
    And user clicks on Search button 
  Then Search is executed  
    And User is redirected to /publications/dictionaries/cancer-terms/def/lung-cancer 
  Then the search bar on the page does not maintain the user’s term 
    And “Starts with” is selected by default 
```

### Resources:
* Feature: JIRA Ticket #
* Web Page: *Here goes a URL to an existing web page*

### Notes

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by sarinapadilla**
* Update TermList to redirect to Definition if search returns one term
* Add unit tests
* Add integration tests and mock data
* Closes #26

### CODE REVIEW

File: cypress/integration/common/search.js [hunk 1]
```diff
@@ -63,3 +63,11 @@ Then('the page contains meta tags with the following names', dataTable => {
     cy.get(locator).should('have.attr', 'content').and('be.eq', content);
   }
 });
+
+Then('the user is redirected to {string}', (redirectUrl) => {
+  cy.location('pathname').should('include', redirectUrl);
+});
+
+Then('the search bar on the page does not maintain the user’s term', () => {
+  cy.get('#keywords').should('have.attr','placeholder').and('be.eq','Enter keywords or phrases');
```
Comment (kate-mashkina) [inline cypress/integration/common/search.js:72]: The placeholder will always be equal to 'Enter keywords or phrases', whether it is displayed or not. You should check the attribute 'value' instead and assert that its empty

File: src/views/Terms/TermList.jsx [hunk 2]
```diff
@@ -15,13 +16,25 @@ const TermList = ({ matchType, query, type }) => {
         : getExpandCharResults(query);
     const { loading, payload } = useCustomQuery( queryAction );
     const [{ language }] = useStateValue();
+    const { DefinitionPath } = useAppPaths();
+    const navigate = useNavigate();
+
+    useEffect( () => {
+        if ( payload && payload.results && payload.results.length === 1 ) {
+            const idOrName = payload.results[0].prettyUrlName
+                ? payload.results[0].prettyUrlName :
```
[the remark this answers is not in the document]
Comment (sarinapadilla) [reply src/views/Terms/TermList.jsx]: Updated in the recently-pushed commit!

File: cypress/integration/common/search.js (hunk 1 above)
Comment (sarinapadilla) [reply cypress/integration/common/search.js]: This makes sense - though there are many other places that we have that placeholder text test. Maybe we should have a ticket to clean those up in the future? But this particular instance has been updated in the recently-pushed commit!

**Comment by VictoriaSunNIH** [comment]
Looks good to me!!

File: cypress/integration/common/search.js (hunk 1 above)
Comment (seyilonge-nci) [reply cypress/integration/common/search.js]: Those instances with `placeholder` text would only be when we are actually checking for `placeholder` values. For this test case scenario we are checking the `input` value which is different.

**Comment by amydugan** [comment]
Removed "Needs UX Review" and added "Passed UX Review" per Victoria.

**Pull request merged**